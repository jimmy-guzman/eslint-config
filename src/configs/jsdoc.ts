import jsdocPlugin from "eslint-plugin-jsdoc";

import { jsdocRules } from "../rules/jsdoc";

export const jsdocConfig = () => {
  return [
    {
      ...jsdocPlugin.configs["flat/recommended-typescript-error"],
      name: "jimmy.codes/jsdoc",
      rules: jsdocRules(),
    },
  ];
};
