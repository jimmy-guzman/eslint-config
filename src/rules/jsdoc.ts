import jsdocPlugin from "eslint-plugin-jsdoc";

import type { Rules } from "../types";

export const jsdocRules = () => {
  return {
    ...jsdocPlugin.configs["flat/recommended-typescript-error"].rules,
    "jsdoc/prefer-import-tag": "error", // Prefer @import over @see for module references
    "jsdoc/require-jsdoc": "off", // Disable enforced JSDoc (TypeScript provides type info)
    "jsdoc/require-next-description": "error", // Ensure @next has a description
    "jsdoc/require-param": "off", // Avoid redundant param documentation (rely on TypeScript)
    "jsdoc/require-returns": "off", // Avoid redundant return type documentation (rely on TypeScript)
    "jsdoc/require-throws-description": "error", // Ensure @throws has a description
    "jsdoc/require-yields-description": "error", // Ensure @yields has a description
    "jsdoc/tag-lines": [
      "error",
      "always",
      {
        applyToEndTag: false, // No need for a blank line before closing JSDoc
        startLines: 1, // Require exactly one blank line before the first tag
      },
    ],
  } satisfies Rules;
};
