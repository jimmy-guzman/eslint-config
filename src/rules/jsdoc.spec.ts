import { jsdocRules } from "./jsdoc";

test("should create jsdoc rules", () => {
  expect(jsdocRules()).toMatchSnapshot();
});
