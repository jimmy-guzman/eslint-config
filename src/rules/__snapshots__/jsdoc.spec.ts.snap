// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should create jsdoc rules 1`] = `
{
  "jsdoc/check-access": "error",
  "jsdoc/check-alignment": "error",
  "jsdoc/check-examples": "off",
  "jsdoc/check-indentation": "off",
  "jsdoc/check-line-alignment": "off",
  "jsdoc/check-param-names": "error",
  "jsdoc/check-property-names": "error",
  "jsdoc/check-syntax": "off",
  "jsdoc/check-tag-names": [
    "error",
    {
      "typed": true,
    },
  ],
  "jsdoc/check-template-names": "off",
  "jsdoc/check-types": "error",
  "jsdoc/check-values": "error",
  "jsdoc/convert-to-jsdoc-comments": "off",
  "jsdoc/empty-tags": "error",
  "jsdoc/escape-inline-tags": "error",
  "jsdoc/implements-on-classes": "error",
  "jsdoc/imports-as-dependencies": "off",
  "jsdoc/informative-docs": "off",
  "jsdoc/lines-before-block": "off",
  "jsdoc/match-description": "off",
  "jsdoc/match-name": "off",
  "jsdoc/multiline-blocks": "error",
  "jsdoc/no-bad-blocks": "off",
  "jsdoc/no-blank-block-descriptions": "off",
  "jsdoc/no-blank-blocks": "off",
  "jsdoc/no-defaults": "error",
  "jsdoc/no-missing-syntax": "off",
  "jsdoc/no-multi-asterisks": "error",
  "jsdoc/no-restricted-syntax": "off",
  "jsdoc/no-types": "error",
  "jsdoc/no-undefined-types": "off",
  "jsdoc/prefer-import-tag": "error",
  "jsdoc/reject-any-type": "error",
  "jsdoc/reject-function-type": "error",
  "jsdoc/require-asterisk-prefix": "off",
  "jsdoc/require-description": "off",
  "jsdoc/require-description-complete-sentence": "off",
  "jsdoc/require-example": "off",
  "jsdoc/require-file-overview": "off",
  "jsdoc/require-hyphen-before-param-description": "off",
  "jsdoc/require-jsdoc": "off",
  "jsdoc/require-next-description": "error",
  "jsdoc/require-next-type": "error",
  "jsdoc/require-param": "off",
  "jsdoc/require-param-description": "error",
  "jsdoc/require-param-name": "error",
  "jsdoc/require-param-type": "off",
  "jsdoc/require-property": "error",
  "jsdoc/require-property-description": "error",
  "jsdoc/require-property-name": "error",
  "jsdoc/require-property-type": "off",
  "jsdoc/require-returns": "off",
  "jsdoc/require-returns-check": "error",
  "jsdoc/require-returns-description": "error",
  "jsdoc/require-returns-type": "off",
  "jsdoc/require-tags": "off",
  "jsdoc/require-template": "off",
  "jsdoc/require-template-description": "error",
  "jsdoc/require-throws": "off",
  "jsdoc/require-throws-description": "error",
  "jsdoc/require-throws-type": "error",
  "jsdoc/require-yields": "error",
  "jsdoc/require-yields-check": "error",
  "jsdoc/require-yields-description": "error",
  "jsdoc/require-yields-type": "error",
  "jsdoc/sort-tags": "off",
  "jsdoc/tag-lines": [
    "error",
    "always",
    {
      "applyToEndTag": false,
      "startLines": 1,
    },
  ],
  "jsdoc/text-escaping": "off",
  "jsdoc/type-formatting": "off",
  "jsdoc/valid-types": "error",
}
`;
